src/store.c

```diff
@@ -30,6 +30,23 @@ struct store_ctx {
 	int expect;
 };
 
+#define MAX_PIN		64
+static char *pin_from_cb(OSSL_PASSPHRASE_CALLBACK *pw_cb, void *pw_cbarg)
+{
+	char cbpin[MAX_PIN + 1] = { 0 };
+	size_t cbpin_len;
+	char *rv;
+
+	if ((pw_cb == NULL) ||
+	    (pw_cb(cbpin, MAX_PIN, &cbpin_len, NULL, pw_cbarg) != OSSL_RV_OK))
```

Comment:
I guess you should pass a param array with OSSL_PASSPHRASE_PARAM_INFO (UTF-8 string) included with a login-info string that describes for what the passphrase is that you are prompting for, e.g. something like _"PKCS#11 USER pin (Slot <n> - <label>)"_. 

```
                OSSL_PARAM params[2] = {
                    OSSL_PARAM_DEFN(OSSL_PASSPHRASE_PARAM_INFO,
                                    OSSL_PARAM_UTF8_STRING, (void *)login_info,
                                    strlen(login_info)),
                    OSSL_PARAM_END,
                };
```

This string will then be part of the password prompt so that the user know for what he is about to enter the passphrase for. There might be several prompts for different tokens/slots, so the user must at least know for which slot/token to enter the pin for.